### PR TITLE
fix(explore): preserve metric column order in bar chart

### PR DESCRIPTION
**Pull request by duynguyenhoang**
### SUMMARY
Reindex pivot table by using metric columns

### TEST PLAN
Create metrics with different name and re-order them, the column in bar chart should reflect those orders

### ADDITIONAL INFORMATION

- [x] Has associated issue: #12416
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by junlincc** [comment]
@adam-stasiak please help testing this one as well

@duynguyenhoang thanks for the PR!!

File: tests/viz_tests.py [hunk 1]
```diff
@@ -531,6 +531,92 @@ def test_column_nulls(self):
...
+            },
+            {
+                "key": "a_column, {}".format(NULL_STRING),
+                "values": [{"x": "pepperoni", "y": 2}, {"x": "cheese", "y": 1}],
+            },
+            {
+                "key": "a_column, engineer",
+                "values": [{"x": "pepperoni", "y": 5}, {"x": "cheese", "y": 3}],
+            },
+        ]
```
Comment (villebro) [inline tests/viz_tests.py:616]: nit: let's use f-strings here
```suggestion
        expected = [
            {
                "key": f"z_column, {NULL_STRING}",
                "values": [{"x": "pepperoni", "y": 2}, {"x": "cheese", "y": 1}],
            },
            {
                "key": "z_column, engineer",
                "values": [{"x": "pepperoni", "y": 5}, {"x": "cheese", "y": 3}],
            },
            {
                "key": f"votes, {NULL_STRING}",
                "values": [{"x": "pepperoni", "y": 2}, {"x": "cheese", "y": 1}],
            },
            {
                "key": "votes, engineer",
                "values": [{"x": "pepperoni", "y": 5}, {"x": "cheese", "y": 3}],
            },
            {
                "key": f"a_column, {NULL_STRING}",
                "values": [{"x": "pepperoni", "y": 2}, {"x": "cheese", "y": 1}],
            },
            {
                "key": "a_column, engineer",
                "values": [{"x": "pepperoni", "y": 5}, {"x": "cheese", "y": 3}],
            },
        ]
```

**Review by villebro — APPROVED** [review]
LGTM, minor style nit

**Comment by duynguyenhoang** [comment]
@villebro Update to f-strings and formated viz_test using black.

**Review by junlincc — APPROVED** [review]
approving as product/qa sign-off.  ✅  thank you so much for your contribution! @duynguyenhoang 

Screen Shot 2021-01-28 at 12 35 51 AM

**Comment by PowerPlop** [comment]
@junlincc @duynguyenhoang  it seems that this order is not respected in the time series bar chart. Is this expected behaviour?

**Pull request merged**